packages/twenty-front/src/pages/onboarding/PaymentSuccess.tsx

```diff
@@ -36,16 +40,32 @@ const StyledButtonContainer = styled.div`
 export const PaymentSuccess = () => {
   const theme = useTheme();
   const currentUser = useRecoilValue(currentUserState);
+  const subscriptionStatus = useSubscriptionStatus();
+  const [getCurrentUser] = useGetCurrentUserLazyQuery();
+  const setCurrentUser = useSetRecoilState(currentUserState);
   const color =
     theme.name === 'light' ? theme.grayScale.gray90 : theme.grayScale.gray10;
 
+  const checkSubscriptionStatus = async () => {
+    if (!isDefined(subscriptionStatus)) {
+      const result = await getCurrentUser({ fetchPolicy: 'network-only' });
+      const currentUser = result.data?.currentUser;
+      if (isDefined(currentUser)) {
```

Comment:
let's make a deep equal here (make it a useRecoilCallback and check the currentUser in the snapshot, this will avoid unnecessary re-renders)